stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt

```diff
@@ -122,7 +127,13 @@ internal class ChatSocketServiceImpl constructor(
             ChatErrorCode.UNABLE_TO_PARSE_SOCKET_EVENT.code,
             ChatErrorCode.NO_ERROR_BODY.code,
             -> {
-                // Nothing to do on that case
+                if (!networkStateProvider.isConnected() && reconnectionAttempts < 3) {
```

Comment:
It's better not to expand the complexity of this class by this logic. We already have `HealthMonitor` with similar logic, we can modify it to cover our case.

Reply:
Move this magic number to constants

Reply:
`by this logic` do you mean `!networkStateProvider.isConnected()`?

Reply:
Also retries condition

Reply:
This way `HealthMonitor` will start to be too much specific and handle logic looks outside of its scope.